src/main/connect/connectionManager.ts

```diff
@@ -562,69 +563,66 @@ export class ConnectionManager implements ExtensionComponent, vscode.Disposable
+     * Password and access token are saved in SecretStorage with an account that is a hash of url and another string.
      * For password - username, access token - a constant.
      * @param keyPair - The second string of the account as described above.
      * @returns The secret if found.
      */
-    private async getSecretFromKeyStore(keyPair: string): Promise<string> {
+    private async getSecretFromSecretStorage(keyPair: string): Promise<string> {
         return (
+            (await this._context.secrets.get(this.createSecretStoreId(this._url, keyPair))) ||
+            (await this._context.secrets.get(this.createSecretStoreId(this._xrayUrl, keyPair))) ||
+            // Deprecated
             (await keytar.getPassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._url, keyPair))) ||
+            // Deprecated
             (await keytar.getPassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._xrayUrl, keyPair))) ||
             ''
         );
     }
 
-    private async deletePasswordFromKeyStore(): Promise<boolean> {
+    private async deletePasswordFromSecretStorage(): Promise<void> {
         if (!this._password) {
-            return true;
+            return;
         }
-        return await this.deleteSecretFromKeyStore(this._username, 'password');
+        await this.deleteSecretFromSecretStorage(this._username);
     }
 
-    private async deleteAccessTokenFromKeyStore(): Promise<boolean> {
+    private async deleteAccessTokenFromSecretStorage(): Promise<void> {
         if (!this._accessToken) {
-            return true;
+            return;
         }
-        return await this.deleteSecretFromKeyStore(ConnectionManager.ACCESS_TOKEN_FS_KEY, 'access token');
+        await this.deleteSecretFromSecretStorage(ConnectionManager.ACCESS_TOKEN_FS_KEY);
     }
 
-    private async deleteSecretFromKeyStore(keyPair: string, secretName: string): Promise<boolean> {
-        let ok: boolean = await keytar.deletePassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._xrayUrl, keyPair));
-        if (!ok) {
-            this._logManager.logMessage('Failed to delete the ' + secretName + ' from the system secrets manager', 'WARN');
-            return false;
-        }
-        return true;
+    private async deleteSecretFromSecretStorage(keyPair: string): Promise<void> {
+        await this._context.secrets.delete(this.createSecretStoreId(this._xrayUrl, keyPair));
+        // Deprecated
+        await keytar.deletePassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._xrayUrl, keyPair));
     }
 
-    private async storePassword() {
+    private async storePassword(): Promise<void> {
         if (!this._password) {
             return;
         }
-        await keytar.setPassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._xrayUrl, this._username), this._password);
+        await this._context.secrets.store(this.createSecretStoreId(this._xrayUrl, this._username), this._password);
```

Comment:
I suggest we use the platform URL instead.

Reply:
Unlike Xray URL, The platform URL exists in less than 100% of the cases